stdlib/tempfile.pyi

```diff
@@ -178,23 +178,22 @@ class _TemporaryFileWrapper(IO[str]):
     # These methods don't exist directly on this object, but
     # are delegated to the underlying IO object through __getattr__.
     # We need to add them here so that this class is concrete.
-    def __iter__(self) -> Iterator[str]: ...
-    def __next__(self) -> str: ...
+    def __iter__(self) -> Iterator[AnyStr]: ...
+    def __next__(self) -> AnyStr: ...
```

Comment:
I think we should delete `__next__`. Tempfiles are not iterators:

```
>>> import tempfile
>>> foo = tempfile.NamedTemporaryFile()
>>> next(foo)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '_TemporaryFileWrapper' object is not an iterator
```

This is a bit confusing, because `_TemporaryFileWrapper`s have a `__next__` attribute, even though they aren't iterators:

```
>>> foo.__next__
<function BufferedRandom.__next__ at 0x7f078294e5f0>
```

The built-in `next()` function and other things don't see this, because it's defined with a `__getattr__` hack, not directly in `class _TemporaryFileWrapper`.

But `__iter__` is different: `_TemporaryFileWrapper` objects are iterable, because `__iter__` is defined in `class _TemporaryFileWrapper`.

```
>>> foo.__iter__
<bound method _TemporaryFileWrapper.__iter__ of <tempfile._TemporaryFileWrapper object at 0x7f0782c81870>>
```

Reply:
Now that I think about it, I think we should add a comment. Something like `# no __next__ method here because obj.__next__() works next(obj) doesn't`

Reply:
Done :)